(core dumped) seqwish

Hi,  I am trying to use seqwish to construct variation graph from raw long reads. It run successfully when using 10,000 reads, however, It  failed when running 100,000 reads, these reads are from the same fastq file. Here are the commands and error information:
`
prefix=test;
minimap2 -x ava-ont -t 48 -c -X $prefix.fq $prefix.fq >$prefix.paf; 
seqwish -t 32 -s $prefix.fq -p $prefix.paf -g $prefix.gfa
`
length for 9b3413df-02a4-423e-8bd0-bb1e69cf7a93, expected 450 but got 0
seqwish: /home/software/seqwish-0.1/src/gfa.cpp:133: void seqwish::emit_gfa(std::ostream&, size_t, const string&, mmmulti::iitree<long unsigned int, long unsigned int>&, mmmulti::iitree<long unsigned int, long unsigned int>&, const sdsl::sd_vector<>&, const rank_1_type&, const select_1_type&, seqwish::seqindex_t&, mmmulti::set<std::pair<long unsigned int, long unsigned int> >&): Assertion `false' failed.
work.sh: line 4:  7666 Aborted                 (core dumped) seqwish -t 32 -s $prefix.fq -p $prefix.paf -g $prefix.gfa
Command exited with non-zero status 134

It seems  maybe there is something wrong with this read '9b3413df-02a4-423e-8bd0-bb1e69cf7a93' in test.fq, here is the sequence and quality score for this read:
`@9b3413df-02a4-423e-8bd0-bb1e69cf7a93
CGTACTTCGTTCAGTTACGTATTGCTAAGGGTTAAAATCAAGACTCGCTGTGCCTAGTTCAGCACCTGTTTCCCACTGGAGGATAATGGGACGCCAGTTTCGAAGGAAACGTTGTTGGGATACTACCCCCACTTGTGTTATAGCCTCTAACCCGGGTAGGTGATCCCTATCGGAGACAGTGTCTGACAGGCAGTTTGACTGGGGCAGTCGCCTCTAAAAGGTAACAGGAGAGCCCAAAGGTTCCCTCAGAATGGTTGGAAGTAATTGCAGTGTAAAGATGTCAGGGCTTGACTGAGCTACAACTCGAGCAGGGACGAAAAGAGATCGAACAGTGATCCGGTGGTTCCGTATGGAAGACCGTACTCAGCAGGATCAAATGCCGTGTCTCAGTTGGAAGCAGGTGCTGAACTGAAGCTGGCTTGAGTCTTGGTTTAACATGGCAATACGTAA
+
&%,$(<9?BI7E1?02.))-++.,784*'543/35053?>0=9EB>:79.(.9.6,0-;8.8=>?BBB1E?<><1;<@:2:;324477?C?;=AK5D?H<6&%'&1#%&),0:=;809>?96))-12,&$##&&2/':&,((&##$%%)&()$+('&))'55-338())%/)022744<@96;?;($$/4114CAD>=ACA=5.((8>>?>;)?1C?;)(62B-''3($*-*)7%18*/?KAA>A,C:34CGI>G?5&)*)**%%(###%.$+3-%'$##%$&&(())*)%$##&)//-9<7?6?820++@A:B>?CA0%%$'-2;(%'$%)0??565:;D>;H;<;0FGIH=64/%43,,)%(0&&)*+/($&'*)##%'%%$&/'0$&#%++*.$$$(&92')8>@=;/-*)%'%2013<=@<>@>),7=6+'&$$#./5+)):977%`

Any help would be appreciated.

> And please let me know which work or not.
> […](#)
> On Sun, Mar 22, 2020, 15:40 Erik Garrison ***@***.***> wrote: Try v0.2. If not that, then try the current master. On Sun, Mar 22, 2020, 15:34 Vincent ***@***.***> wrote: > You've found an error that I'm aware of. Sorry for this. You can > downgrade and get a much lower-performance version of seqwish that should > induce the same graph. > > I have a test case and I need to use it to fix this. If you can share > yours in some way then I'd be very appreciative as well. > > Thank you for your quick response. But I use release version of v0.1 of > seqwish from here: > https://github.com/ekg/seqwish/releases/tag/v0.1 . Should I try v0.2 ? > > — > You are receiving this because you commented. > Reply to this email directly, view it on GitHub > <[#36 (comment)](https://github.com/ekg/seqwish/issues/36#issuecomment-602215842)>, or > unsubscribe > <https://github.com/notifications/unsubscribe-auth/AABDQEKFOL22VSGTPXKD3TTRIYOXHANCNFSM4LRJ4TJA> > . >

Hi, I tried both v0.2 and the current master, but it failed when building, I could not figure it out. Could you please help me? here is the error info:

`cmake -H. -Bbuild && cmake --build build -- -j3
`
-- The C compiler identification is GNU 7.3.0
-- The CXX compiler identification is GNU 7.3.0
-- Check for working C compiler: /export/scratch1/home/vincent/software/miniconda3/bin/x86_64-conda_cos6-linux-gnu-cc
-- Check for working C compiler: /export/scratch1/home/vincent/software/miniconda3/bin/x86_64-conda_cos6-linux-gnu-cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /export/scratch1/home/vincent/software/miniconda3/bin/x86_64-conda_cos6-linux-gnu-c++
-- Check for working CXX compiler: /export/scratch1/home/vincent/software/miniconda3/bin/x86_64-conda_cos6-linux-gnu-c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
CMake Error at /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525 (message):
  No download info given for 'sdsl-lite' and its source directory:

   /export/scratch3/vincent/software/seqwish-0.2/deps/sdsl-lite

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100 (_ep_add_download_command)
  CMakeLists.txt:53 (ExternalProject_Add)


CMake Error at /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525 (message):
  No download info given for 'tayweeargs' and its source directory:

   /export/scratch3/vincent/software/seqwish-0.2/deps/args

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100 (_ep_add_download_command)
  CMakeLists.txt:65 (ExternalProject_Add)


CMake Error at /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525 (message):
  No download info given for 'gzipreader' and its source directory:

   /export/scratch3/vincent/software/seqwish-0.2/deps/gzip_reader

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100 (_ep_add_download_command)
  CMakeLists.txt:73 (ExternalProject_Add)


CMake Error at /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525 (message):
  No download info given for 'mmmultimap' and its source directory:

   /export/scratch3/vincent/software/seqwish-0.2/deps/mmmultimap

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100 (_ep_add_download_command)
  CMakeLists.txt:81 (ExternalProject_Add)


CMake Error at /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525 (message):
  No download info given for 'iitii' and its source directory:

   /export/scratch3/vincent/software/seqwish-0.2/deps/iitii

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100 (_ep_add_download_command)
  CMakeLists.txt:90 (ExternalProject_Add)


CMake Error at /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525 (message):
  No download info given for 'mmap_allocator' and its source directory:

   /export/scratch3/vincent/software/seqwish-0.2/deps/mmap_allocator

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100 (_ep_add_download_command)
  CMakeLists.txt:97 (ExternalProject_Add)


CMake Error at /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525 (message):
  No download info given for 'ips4o' and its source directory:

   /export/scratch3/vincent/software/seqwish-0.2/deps/ips4o

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100 (_ep_add_download_command)
  CMakeLists.txt:106 (ExternalProject_Add)


CMake Error at /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525 (message):
  No download info given for 'bbhash' and its source directory:

   /export/scratch3/vincent/software/seqwish-0.2/deps/BBHash

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100 (_ep_add_download_command)
  CMakeLists.txt:115 (ExternalProject_Add)


CMake Error at /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525 (message):
  No download info given for 'atomicbitvector' and its source directory:

   /export/scratch3/vincent/software/seqwish-0.2/deps/atomicbitvector

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100 (_ep_add_download_command)
  CMakeLists.txt:125 (ExternalProject_Add)


CMake Error at /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525 (message):
  No download info given for 'atomicqueue' and its source directory:

   /export/scratch3/vincent/software/seqwish-0.2/deps/atomic_queue

  is not an existing non-empty directory.  Please specify one of:

   * SOURCE_DIR with an existing non-empty directory
   * DOWNLOAD_COMMAND
   * URL
   * GIT_REPOSITORY
   * SVN_REPOSITORY
   * HG_REPOSITORY
   * CVS_REPOSITORY and CVS_MODULE
Call Stack (most recent call first):
  /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100 (_ep_add_download_command)
  CMakeLists.txt:135 (ExternalProject_Add)


-- Configuring incomplete, errors occurred!
See also "/export/scratch3/vincent/software/seqwish-0.2/build/CMakeFiles/CMakeOutput.log".

git clone --recursive

or to avoid pulling again

git submodules update --init --recursive
git submodules sync
git submodules update --init --recursive

On Sun, Mar 22, 2020 at 8:31 PM Vincent <notifications@github.com> wrote:

> And please let me know which work or not.
> … <#m_8859744613341929828_>
> On Sun, Mar 22, 2020, 15:40 Erik Garrison *@*.*> wrote: Try v0.2. If not
> that, then try the current master. On Sun, Mar 22, 2020, 15:34 Vincent @.*>
> wrote: > You've found an error that I'm aware of. Sorry for this. You can >
> downgrade and get a much lower-performance version of seqwish that should >
> induce the same graph. > > I have a test case and I need to use it to fix
> this. If you can share > yours in some way then I'd be very appreciative as
> well. > > Thank you for your quick response. But I use release version of
> v0.1 of > seqwish from here: >
> https://github.com/ekg/seqwish/releases/tag/v0.1 . Should I try v0.2 ? >
> > — > You are receiving this because you commented. > Reply to this email
> directly, view it on GitHub > <#36 (comment)
> <https://github.com/ekg/seqwish/issues/36#issuecomment-602215842>>, or >
> unsubscribe >
> https://github.com/notifications/unsubscribe-auth/AABDQEKFOL22VSGTPXKD3TTRIYOXHANCNFSM4LRJ4TJA
> > . >
>
> Hi, I tried both v0.2 and the current master, but it failed when building,
> I could not figure it out. Could you please help me? here is the error info:
>
> cmake -H. -Bbuild && cmake --build build -- -j3
> -- The C compiler identification is GNU 7.3.0
> -- The CXX compiler identification is GNU 7.3.0
> -- Check for working C compiler:
> /export/scratch1/home/vincent/software/miniconda3/bin/x86_64-conda_cos6-linux-gnu-cc
> -- Check for working C compiler:
> /export/scratch1/home/vincent/software/miniconda3/bin/x86_64-conda_cos6-linux-gnu-cc
> -- works
> -- Detecting C compiler ABI info
> -- Detecting C compiler ABI info - done
> -- Detecting C compile features
> -- Detecting C compile features - done
> -- Check for working CXX compiler:
> /export/scratch1/home/vincent/software/miniconda3/bin/x86_64-conda_cos6-linux-gnu-c++
> -- Check for working CXX compiler:
> /export/scratch1/home/vincent/software/miniconda3/bin/x86_64-conda_cos6-linux-gnu-c++
> -- works
> -- Detecting CXX compiler ABI info
> -- Detecting CXX compiler ABI info - done
> -- Detecting CXX compile features
> -- Detecting CXX compile features - done
> -- Found OpenMP_C: -fopenmp (found version "4.5")
> -- Found OpenMP_CXX: -fopenmp (found version "4.5")
> -- Found OpenMP: TRUE (found version "4.5")
> CMake Error at
> /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525
> (message):
> No download info given for 'sdsl-lite' and its source directory:
>
> /export/scratch3/vincent/software/seqwish-0.2/deps/sdsl-lite
>
> is not an existing non-empty directory. Please specify one of:
>
>    - SOURCE_DIR with an existing non-empty directory
>    - DOWNLOAD_COMMAND
>    - URL
>    - GIT_REPOSITORY
>    - SVN_REPOSITORY
>    - HG_REPOSITORY
>    - CVS_REPOSITORY and CVS_MODULE
>    Call Stack (most recent call first):
>    /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100
>    (_ep_add_download_command)
>    CMakeLists.txt:53 (ExternalProject_Add)
>
> CMake Error at
> /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525
> (message):
> No download info given for 'tayweeargs' and its source directory:
>
> /export/scratch3/vincent/software/seqwish-0.2/deps/args
>
> is not an existing non-empty directory. Please specify one of:
>
>    - SOURCE_DIR with an existing non-empty directory
>    - DOWNLOAD_COMMAND
>    - URL
>    - GIT_REPOSITORY
>    - SVN_REPOSITORY
>    - HG_REPOSITORY
>    - CVS_REPOSITORY and CVS_MODULE
>    Call Stack (most recent call first):
>    /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100
>    (_ep_add_download_command)
>    CMakeLists.txt:65 (ExternalProject_Add)
>
> CMake Error at
> /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525
> (message):
> No download info given for 'gzipreader' and its source directory:
>
> /export/scratch3/vincent/software/seqwish-0.2/deps/gzip_reader
>
> is not an existing non-empty directory. Please specify one of:
>
>    - SOURCE_DIR with an existing non-empty directory
>    - DOWNLOAD_COMMAND
>    - URL
>    - GIT_REPOSITORY
>    - SVN_REPOSITORY
>    - HG_REPOSITORY
>    - CVS_REPOSITORY and CVS_MODULE
>    Call Stack (most recent call first):
>    /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100
>    (_ep_add_download_command)
>    CMakeLists.txt:73 (ExternalProject_Add)
>
> CMake Error at
> /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525
> (message):
> No download info given for 'mmmultimap' and its source directory:
>
> /export/scratch3/vincent/software/seqwish-0.2/deps/mmmultimap
>
> is not an existing non-empty directory. Please specify one of:
>
>    - SOURCE_DIR with an existing non-empty directory
>    - DOWNLOAD_COMMAND
>    - URL
>    - GIT_REPOSITORY
>    - SVN_REPOSITORY
>    - HG_REPOSITORY
>    - CVS_REPOSITORY and CVS_MODULE
>    Call Stack (most recent call first):
>    /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100
>    (_ep_add_download_command)
>    CMakeLists.txt:81 (ExternalProject_Add)
>
> CMake Error at
> /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525
> (message):
> No download info given for 'iitii' and its source directory:
>
> /export/scratch3/vincent/software/seqwish-0.2/deps/iitii
>
> is not an existing non-empty directory. Please specify one of:
>
>    - SOURCE_DIR with an existing non-empty directory
>    - DOWNLOAD_COMMAND
>    - URL
>    - GIT_REPOSITORY
>    - SVN_REPOSITORY
>    - HG_REPOSITORY
>    - CVS_REPOSITORY and CVS_MODULE
>    Call Stack (most recent call first):
>    /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100
>    (_ep_add_download_command)
>    CMakeLists.txt:90 (ExternalProject_Add)
>
> CMake Error at
> /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525
> (message):
> No download info given for 'mmap_allocator' and its source directory:
>
> /export/scratch3/vincent/software/seqwish-0.2/deps/mmap_allocator
>
> is not an existing non-empty directory. Please specify one of:
>
>    - SOURCE_DIR with an existing non-empty directory
>    - DOWNLOAD_COMMAND
>    - URL
>    - GIT_REPOSITORY
>    - SVN_REPOSITORY
>    - HG_REPOSITORY
>    - CVS_REPOSITORY and CVS_MODULE
>    Call Stack (most recent call first):
>    /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100
>    (_ep_add_download_command)
>    CMakeLists.txt:97 (ExternalProject_Add)
>
> CMake Error at
> /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525
> (message):
> No download info given for 'ips4o' and its source directory:
>
> /export/scratch3/vincent/software/seqwish-0.2/deps/ips4o
>
> is not an existing non-empty directory. Please specify one of:
>
>    - SOURCE_DIR with an existing non-empty directory
>    - DOWNLOAD_COMMAND
>    - URL
>    - GIT_REPOSITORY
>    - SVN_REPOSITORY
>    - HG_REPOSITORY
>    - CVS_REPOSITORY and CVS_MODULE
>    Call Stack (most recent call first):
>    /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100
>    (_ep_add_download_command)
>    CMakeLists.txt:106 (ExternalProject_Add)
>
> CMake Error at
> /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525
> (message):
> No download info given for 'bbhash' and its source directory:
>
> /export/scratch3/vincent/software/seqwish-0.2/deps/BBHash
>
> is not an existing non-empty directory. Please specify one of:
>
>    - SOURCE_DIR with an existing non-empty directory
>    - DOWNLOAD_COMMAND
>    - URL
>    - GIT_REPOSITORY
>    - SVN_REPOSITORY
>    - HG_REPOSITORY
>    - CVS_REPOSITORY and CVS_MODULE
>    Call Stack (most recent call first):
>    /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100
>    (_ep_add_download_command)
>    CMakeLists.txt:115 (ExternalProject_Add)
>
> CMake Error at
> /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525
> (message):
> No download info given for 'atomicbitvector' and its source directory:
>
> /export/scratch3/vincent/software/seqwish-0.2/deps/atomicbitvector
>
> is not an existing non-empty directory. Please specify one of:
>
>    - SOURCE_DIR with an existing non-empty directory
>    - DOWNLOAD_COMMAND
>    - URL
>    - GIT_REPOSITORY
>    - SVN_REPOSITORY
>    - HG_REPOSITORY
>    - CVS_REPOSITORY and CVS_MODULE
>    Call Stack (most recent call first):
>    /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100
>    (_ep_add_download_command)
>    CMakeLists.txt:125 (ExternalProject_Add)
>
> CMake Error at
> /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:2525
> (message):
> No download info given for 'atomicqueue' and its source directory:
>
> /export/scratch3/vincent/software/seqwish-0.2/deps/atomic_queue
>
> is not an existing non-empty directory. Please specify one of:
>
>    - SOURCE_DIR with an existing non-empty directory
>    - DOWNLOAD_COMMAND
>    - URL
>    - GIT_REPOSITORY
>    - SVN_REPOSITORY
>    - HG_REPOSITORY
>    - CVS_REPOSITORY and CVS_MODULE
>    Call Stack (most recent call first):
>    /export/scratch3/vincent/software/miniconda3/share/cmake-3.12/Modules/ExternalProject.cmake:3100
>    (_ep_add_download_command)
>    CMakeLists.txt:135 (ExternalProject_Add)
>
> -- Configuring incomplete, errors occurred!
> See also
> "/export/scratch3/vincent/software/seqwish-0.2/build/CMakeFiles/CMakeOutput.log".
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ekg/seqwish/issues/36#issuecomment-602260321>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABDQEO4B2VTQNQ2T3GVTV3RIZRQ5ANCNFSM4LRJ4TJA>
> .
>

--- a/src/transclosure.cpp
+++ b/src/transclosure.cpp
@@ -18,7 +18,7 @@ void extend_range(const uint64_t& s_pos,
     // if one doesn't exist, add the range
     if (f == range_buffer.end()) {
         range_buffer[q_pos] = {s_pos, s_pos+1};
-    } else if (seqidx.seq_start(offset(q_pos))) {
+    } else if (seqidx.seq_start(offset(q_pos)) || (is_rev(q_pos) && seqidx.seq_start(offset(q_last_pos)))) {
         // flush the buffer we found, so we don't extend across node boundaries
         flush_range(f, node_iitree, path_iitree);
         range_buffer.erase(f);

